gpflow/conditionals.py

```diff
@@ -344,17 +374,29 @@ def _sample_mvn(mean, cov, cov_structure):
     - "full": cov holds the full covariance matrix (without jitter)
     :return: sample from the MVN of shape N x D
     """
-    eps = tf.random_normal(tf.shape(mean), dtype=settings.float_type)  # N x P
+    rank_mean = len(mean.get_shape())
```

Comment:
These two lines will not work for unknown shapes. Requires changes to use `tf.equal_assert(..)`.

Reply:
Isn't tensor.get_shape().ndim a property of the static graph? Is the rank of a tensor (the length of the shape) not known at graph-build time?